src/modules/LosslessCompression/Module.js

```diff
@@ -0,0 +1,44 @@
+module.exports = function LosslessCompress(options, UI) {
+  function draw(input, callback, progressObj) {
+    const compress_images = require("compress-images");
+    compress_images(
+      "images/**/*.{jpg,JPG,jpeg,JPEG,png}",
+      "results/",
+      { compress_force: false, statistic: true, autoupdate: true },
+      false,
+      { jpg: { engine: "jpegoptim", command: ["--all-progressive", "-d"] } },
+      { png: { engine: "optipng" } },
+      function(err, completed) {
+        if (completed === true) {
+          var reader = new FileReader();
+          reader.readAsDataURL("results/**/*");
+          reader.onloadend = function() {
+            base64data = reader.result;
+            output(null, base64data, input.format, false);
+            if (callback) callback();
+            return;
+          };
+        }
+      }
+    );
+    progressObj.stop(true);
+    progressObj.overrideFlag = true;
+
+    var step = this;
+
+    function output(image, datauri, mimetype, wasmSuccess) {
+      step.output = {
+        src: datauri,
+        format: mimetype,
+        wasmSuccess,
+        useWasm: options.useWasm
+      };
+    }
+  }
+  return {
+    options: options,
+    draw: draw,
+    output: output,
+    UI: UI
+  };
+};
```

Comment:
does this work? What do you get as an output after running this? @daemon1024

Reply:
I can't seem to insert the lossless-compression as a step. I can't seem to understand how to pass the input in the function. The parameter required there is path to the file, but the incoming data is a dataURI.

Reply:
https://hastebin.com/bakopiwino.coffeescript
This is the error I get when i run `npm run test-all`

Reply:
don't run the test right now. Tests are to be ran later when the module is complete.

Reply:
see from line 64 to line 81 of Module.js file of MinifyImage module inside src/modules/MinifyImage. There they have taken dirname/images as a file path and passed imagemin on it. @daemon1024

Reply:
@niravasher that part in Minify Module seems like to be used when the inBrowser options is false. They created that to run rest and stuff prolly.